src/common/utils/url-hash-converter/url-hash-converter.ts

```diff
@@ -34,17 +34,19 @@ export const urlHashConverter: UrlHashConverter = {
   essenceFromHash(hash: string, dataCube: DataCube, visualizations: Manifest[]): Essence {
     const hashParts = hash.split('/');
 
-    if (hashParts.length < 3)
-      throw new Error("Wrong url hash structure.");
-
     const isLegacyVersionWithVisualisationPrefix =
-      hashParts[1] === LEGACY_VIEW_DEFINITION_VERSION && version2Visualizations.indexOf(hashParts[0]) !== -1;
+      hashParts.length >= 3
+      && hashParts[1] === LEGACY_VIEW_DEFINITION_VERSION
+      && version2Visualizations.indexOf(hashParts[0]) !== -1;
 
     let visualization;
     if (isLegacyVersionWithVisualisationPrefix) {
       visualization = hashParts.shift();
     }
 
+    if (hashParts.length < 2)
```

Comment:
I would move this invariant a the top of method, and add more descriptive message, how many tokens we expect and how many we got.